presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -133,7 +133,7 @@ public void run()
                 Optional.ofNullable(clientOptions.truststorePath),
                 Optional.ofNullable(clientOptions.truststorePassword),
                 Optional.ofNullable(clientOptions.user),
-                Optional.ofNullable(promptFor(clientOptions.password)),
+                getPassword(),
```

Comment:
`getPassword()` does both. If the `-P` is specified - it takes the password from parameter. If the `--password` flag is specified - it shows the prompt.